alternator/executor.cc

```diff
@@ -1772,8 +1772,9 @@ future<executor::request_return_type> executor::execute_rmw_operation(client_sta
             });
         });
     }
-    return op->execute(_proxy, client_state, trace_state, std::move(permit), needs_read_before_write, _stats).finally([op, start_time, this] {
+    return op->execute(_proxy, client_state, trace_state, std::move(permit), needs_read_before_write, _stats).then([op, start_time, this] (executor::request_return_type ret) {
```

Comment:
one thing I'm still wondering is if we should also skip the stats in case there is another kind of errors (the `request_return_type` is still a variant that includes `api_error` - so whether we should also check that and skip the stats in case there is an `api_arror` result)

Reply:
I don't have a strong opinion, but two questions: what would the user experience be? and what about failures,  are you going to count them somehow?

Reply:
@emdroid I agree with Amnon we need to have account for failures as well, just in a different metric. 
Are we doing this already?

Reply:
I couldn't find any yet, at least not in the https://github.com/scylladb/scylladb/blob/master/alternator/stats.hh there doesn't seem to be anything specific to errors/timeouts. But could still be those are being checked somewhere upstream, will keep looking...